File: exercise/php/day23/src/Core/Control/MagicPowerAmplifier.php
<?php

namespace Core\Control;

class MagicPowerAmplifier
{
    private AmplifierType $amplifierType;

    public function __construct(AmplifierType $amplifierType)
    {
        $this->amplifierType = $amplifierType;
    }

    public function amplify(float $magicPower): float
    {
        return $magicPower > 0 ? $magicPower * $this->amplifierType->getMultiplier() : $magicPower;
    }
}

File: exercise/java/day23/src/main/java/core/control/MagicPowerAmplifier.java
package core.control;

public class MagicPowerAmplifier {
    private AmplifierType amplifierType;

    public MagicPowerAmplifier(AmplifierType amplifierType) {
        this.amplifierType = amplifierType;
    }

    public float amplify(float magicPower) {
        if (magicPower > 0)
            return magicPower * amplifierType.getMultiplier();
        return magicPower;
    }
}
